specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-04-01-preview/machineLearningServices.json

```diff
@@ -6112,6 +6388,243 @@
       "additionalProperties": false,
       "x-ms-discriminator-value": "ManagedIdentity"
     },
+    "ManagedNetworkDto": {
+      "type": "object",
+      "properties": {
+        "isolationMode": {
+          "$ref": "#/definitions/IsolationMode"
+        },
+        "outboundRules": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/OutboundRule"
+          }
+        }
+      }
+    },
+    "ManagedNetworkProvisionOptions": {
+      "type": "object",
+      "properties": {
+        "includeSpark": {
+          "type": "boolean"
+        }
+      }
+    },
+    "ManagedNetworkProvisionStatus": {
+      "type": "object",
+      "properties": {
+        "status": {
+          "$ref": "#/definitions/ManagedNetworkStatus"
+        },
+        "sparkReady": {
+          "type": "boolean"
+        }
+      }
+    },
+    "ManagedNetworkSettings": {
+      "type": "object",
+      "properties": {
+        "isolationMode": {
+          "$ref": "#/definitions/IsolationMode"
+        },
+        "networkId": {
+          "type": "string",
+          "readOnly": true
+        },
+        "outboundRules": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/OutboundRule"
+          },
+          "x-nullable": true
+        },
+        "status": {
+          "$ref": "#/definitions/ManagedNetworkProvisionStatus"
+        }
+      }
+    },
+    "ManagedNetworkStatus": {
+      "enum": [
+        "Inactive",
+        "Active"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ManagedNetworkStatus",
+        "modelAsString": true
+      }
+    },
+    "IsolationMode": {
+      "enum": [
+        "Disabled",
+        "AllowInternetOutbound",
+        "AllowOnlyApprovedOutbound"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "IsolationMode",
+        "modelAsString": true
+      }
+    },
+    "OutboundRuleBasicResource": {
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
```

Comment:
Consider using x-ms-client-flatten here, otherwise expected model looks like this:

"properties": {
        "properties": { // See if this extra properties nesting required here
                "type": {},
                "status": {},
                "category": {}
        }
      },

Reply:
done

Reply:
went back on this one. do not flatten as nested properties for outbound rules are needed. flatten produces bad results for pythong autorest SDK restclient generation, needs nested structure.